### PR TITLE
Changes to the Benchmark Suite #2 

### PR DESCRIPTION
**Pull request by Jeriimiah**
Signed-off-by: Jerimiah <Jerimiah.wright@utexas.edu>

Added a toggle to the output logs generated by the benchmark suite. The default is no output log. Also added a return statement for the buffer, so you can interact with the data after running the suite. (xacc.py)

Fixed an initialization issue with the pyscf observable (pyscf_observable.py)

Added pyscf and adapt-vqe and its keywords to the benchmark suite (chemistry_benchmark.py)

### CODE REVIEW

File: python/benchmark/chemistry/chemistry_benchmark.py [hunk 1]
```diff
@@ -36,13 +36,18 @@ def execute(self, inputParams):
         elif inputParams['Observable']['name'] == 'fermion':
             obs_str = inputParams['Observable']['obs_str']
             H = xacc.getObservable('fermion', obs_str)
-
         elif inputParams['Observable']['name'] == 'psi4':
```
Comment (amccaskey) [inline python/benchmark/chemistry/chemistry_benchmark.py:39]: These 2 elifs provide the same code, you should remove the pyscf one and replace the one elif with 
```python
elif inputParams['Observable']['name'] in ['pyscf','psi4']:
```

File: python/plugins/observables/pyscf_observable.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def fromOptions(self, inputParams):
         mol = gto.mole.Mole()
         mol.atom = inputParams['geometry']
         mol.basis = inputParams['basis']
+        mol.build()
```
Comment (amccaskey) [inline python/plugins/observables/pyscf_observable.py:38]: what happened here? why did you need to add this method call? @danclaudino is this needed?

File: python/xacc.py [hunk 3]
```diff
@@ -406,9 +406,19 @@ def benchmark(xacc_settings):
 
     results_name = "%s_%s_out" % (os.path.splitext(
         tail)[0], timestr)
-    f = open(results_name+".ab", 'w')
-    f.write(str(buffer))
-    f.close()
+    
+    if 'output_logs' not in xacc_settings['Benchmark']:
```
Comment (amccaskey) [inline python/xacc.py:410]: I prefer this to work the other way - by default log files should be written, and you can specify not doing it in the input file or input dictionary.

File: python/plugins/observables/pyscf_observable.py (hunk 2 above)
Comment (danclaudino) [reply python/plugins/observables/pyscf_observable.py:38]: `build()` needs to explicitly called only when changing members of `gto.mole.Mole()`, which is not the case here

Comment (Jeriimiah) [reply python/plugins/observables/pyscf_observable.py:38]: Without that line, you get a warning about calling scf before initializing the molecule. It's found in https://sunqm.github.io/pyscf/_modules/pyscf/scf/hf.html, specifically in the SCF class. In there they have an example of how to use the class, and a conditional to add that that mol.build line if it was not already present.

    def __init__(self, mol):
        if not mol._built:
            sys.stderr.write('Warning: %s must be initialized before calling SCF.\n'
                             'Initialize %s in %s\n' % (mol, mol, self))
            mol.build()

Comment (danclaudino) [reply python/plugins/observables/pyscf_observable.py:38]: I just tried and did get a warning. It may be because of a newer version, but does not impact performance or change the expected behavior in anyway, as it does call `build()`. Maybe we can accept this to avoid having this warning show up.

Comment (amccaskey) [reply python/plugins/observables/pyscf_observable.py:38]: cool I vote to accept it

Comment (danclaudino) [reply python/plugins/observables/pyscf_observable.py:38]: I'm cool with it

File: python/benchmark/chemistry/chemistry_benchmark.py (hunk 1 above)
Comment (Jeriimiah) [reply python/benchmark/chemistry/chemistry_benchmark.py]: I fixed this - Do I need to make a new pull request in order for the new changes to take affect?
